var/spack/repos/builtin/packages/hpx/package.py

```diff
@@ -100,6 +101,10 @@ class Hpx(CMakePackage, CudaPackage):
 
     # MPI
     depends_on('mpi', when='networking=mpi')
+    depends_on('mpi', when='+async_mpi')
```

Comment:
Is there any relationship/constraint between `async_mpi` and `networking=mpi`?

Reply:
Yes, there is, `networking=mpi` implies `async_mpi`. I am not sure how to describe the relationship in spack though. Ideally, I would be able to set the default value of `async_mpi` based on whether `networking=mpi`.

Reply:
If it is not a constraint, so any combination of `networking` and `async_mpi` is valid, I think it is perfect like this. 👍